New container generation script for Mageia 6+

Conan-Kudo opened the pull request:
This script enables Mageia containers for Mageia 6+ to be created with the expected default minimal configuration.

It defaults to DNF + urpmi, just like all the other released Mageia 6 software, but allows people to build custom images with their own choice of the three supported package managers for containers.

Since Mageia 6 is formally releasing tomorrow, please use this new script to generate the containers for Mageia 6 to publish to the Docker Hub.

juanluisbaptiste commented:
I suppose the image has to be built on cauldron to have dnf available or has it been backported ? (sorry, I have been away from mga development list from quite some time now so I'm not familiar with dnf).

Conan-Kudo commented:
@juanluisbaptiste The image can be built on Mageia 6 or Cauldron. Unfortunately, it cannot be backported to Mageia 5.

Conan-Kudo commented:
@juanluisbaptiste More details on this new package manager can be found here: https://wiki.mageia.org/en/Mageia_6_Release_Notes#New_alternative_package_manager:_DNF

juanluisbaptiste commented:
Which package manager should be used for the default image ? I suppose the idea is have images for both urpmi and dnf, or the latter is replacing the former ?

juanluisbaptiste commented:
Also I see that mkimage-dnf does nothing with --package-manager=urpmi. I'll work with dnf for now.

juanluisbaptiste commented:
Is there a way to specify a mirror with dnf ? somtimes I'm getting 404 errors with the repositories its choosing.

Conan-Kudo commented:
@juanluisbaptiste There is.

`--disablerepo=* --repofrompath=mgamirror,http://mirrors.kernel.org/mageia/distrib/6/x86_64/media/core/release/ --repofrompath=updatesmirror,http://mirrors.kernel.org/mageia/distrib/6/x86_64/media/core/updates/ --enablerepo=mgamirror --enablerepo=updatesmirror`

Would you like me to add that?

Conan-Kudo commented:
@juanluisbaptiste When you don't specify `--package-manager`, it'll install both `dnf` and `urpmi` into the image.

Conan-Kudo committed:
New DNF-based container generation script for Mageia 6+

This script enables Mageia containers for Mageia 6+ to be created
with the expected default minimal configuration.

It defaults to DNF + urpmi, just like all the other released Mageia 6
software, but allows people to build custom images with their own choice
of the three supported package managers for containers.

Conan-Kudo commented:
@juanluisbaptiste I've added an option to specify a mirror to the script.

juanluisbaptiste commented:
Ok I managed to build it, there's a unary operator error at the end that I have fixed (to be added after I merge the PR). Now I'll update the docker image.

juanluisbaptiste commented:
Ok, tarball uploaded: https://github.com/juanluisbaptiste/docker-brew-mageia/tree/dist/6, @tianon could you please check if everything is ok for this new release ?

juanluisbaptiste commented:
Also @tianon , it seems the process to update the library file has changed since I last updated and I'm a little confused with how it works now. Could you point me in the right direction ?

tianon commented:
@juanluisbaptiste tarball looks good, `Dockerfile` looks good -- branch only has a rootfs for 6 now, is that intended?  Is version 5 still supported?  (They could each live in separate branches if you really wanted -- I'm happy to help restore the previous commit contents; I've still got local copies of all that, as does https://github.com/tianon/bad-ideas/tags.)

The process shouldn't be different -- just a PR which updates https://github.com/docker-library/official-images/blob/master/library/mageia. :+1:

juanluisbaptiste commented:
@tianon yes, mageia 5 is still supported, you mean that in the dist branch on the mageia 6 commit should also have the mageia 5 rootfs tarball ?

tianon commented:
Well, it needs to exist somewhere. :smile:

Having one commit with both tarballs would be fine, two commits one tarball each would be fine, or two branches (which is probably easiest to maintain long-term).

tianon commented:
I've used the pattern of just using `git commit --amend` with a generic message like `Add tarballs` successfully before (then you just keep a local checkout that has a single commit of "added tarballs" constantly).

juanluisbaptiste commented:
> I've used the pattern of just using git commit --amend with a generic message like Add tarballs successfully before (then you just keep a local checkout that has a single commit of "added tarballs" constantly).

@tianon I'm sorry, I'm confused on how to restore the mga 5 tarball, your last comment would be for your first suggestion of one commit with the two tarballs ?

tianon commented:
I think the easiest/quickest way to get the 5 tarball back is to download
it from
https://github.com/juanluisbaptiste/docker-brew-mageia/blob/c01771b53f87dd8d130879122d13025a9d957000/5/rootfs.tar.xz
(which is the commit currently used by the official-images).  Then just use
"git commit --amend" to add it to that existing "dist" commit and you've
got a single commit on top of master which adds two tarballs.

Make sense?

juanluisbaptiste commented:
@tianon done, now I'm updating the library file with the new commit.

juanluisbaptiste commented:
PR here https://github.com/docker-library/official-images/pull/3208

juanluisbaptiste commented:
@Conan-Kudo Images are ready !
